File: src/test/java/com/raychenon/codility/BracketsTest.java
package com.raychenon.codility;

import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 6/1/19
 * Time: 9:30 PM
 * https://codility.com/programmers/task/brackets/
 */
public class BracketsTest {

    @Test
    public void assertBracketsSolution() {
        assertEquals(Brackets.hasBalancedBrackets("{[()()]}"), true);
        assertEquals(Brackets.hasBalancedBrackets("([)>>()]"), false);
        assertEquals(Brackets.hasBalancedBrackets("([)()]"), false);

        assertEquals(Brackets.hasBalancedBrackets("{[()<>()]}"), true);
        // can contain other characters
        assertEquals(Brackets.hasBalancedBrackets("{an}"), true);
    }

    @Test
    public void assertEmptyBracketsSolution() {

        assertEquals(Brackets.hasBalancedBrackets(""), true);
        assertEquals(Brackets.hasBalancedBrackets(null), true);
    }
}

File: src/test/kotlin/com/raychenon/kotlin/codility/BracketsTest.kt
package com.raychenon.kotlin.codility

import com.raychenon.codility.Brackets
import org.junit.Assert
import org.junit.Test

class BracketsTest {

    @Test
    fun evaluateBrackets(){

        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("{}"),true)
        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("{[()()]}"),true)

        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("([)>>()]"), false)
        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("([)()]"), false)

        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("{[()<>()]}"), true)
        // can contain other characters
        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("{an}"), true)
        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets("{any}"), true)
    }

    @Test
    fun evaluateEmptyBrackets(){
        Assert.assertEquals(BracketsKotlin.hasBalancedBrackets(""), true)
    }

    @Test
    fun assertOneSidedBracketsSolution() {
        Assert.assertEquals(Brackets.hasBalancedBrackets("{["), false)
        Assert.assertEquals(Brackets.hasBalancedBrackets("{"), false)
    }

}